### DISCUSSION TITLE
Best/Recommended Filesystem

### DISCUSSION THREAD

**Discussion by Whytey** [blakeblackshear/frigate#1072] [Ask A Question]
I am about to format a new disk to move my frigate clips and recordings to.  Any recommendations what filesystem to use?  For reference my system is showing files of size of ~20MB for recording files and generally around 12MB for clips, with JPG files being under 100KB.  

If nothing, I will likely default to ext4!

Cheers,
Whytey

**Comment by felipecrs** [comment]
I have the same question. I wonder if a different file system is better for a portable USB HDD I only use for Frigate.

**Comment by leccelecce** [comment]
Currently Frigate can put hundreds of thousands of event snapshots into a single directory if you have a high volume of events and/or snapshot retention times. If that applies to you, I'd check before picking any filesystem that it copes OK with that. I'd certainly avoid anything older such as EXT3. I would like to get round to a PR to move to date-stamped folder names, but it's a fair bit of work given inconsistent approaches around the codebase plus the fact you'd have to migrate existing data for users with some careful scripting that might take 5-10 minutes to run on startup, while potentially hammering IO.

Note that there is a significant difference between e.g. EXT4 saying it copes with trillions+ of files in a directory and the actual performance of various tools and libraries that interact with the filesystem. For example, if I try and browse to the snapshots directory with my NAS, it will crash the UI process as it's poorly coded and probably trying to load all metadata at once. Similarly anything that produces thumbnails or indexes directories could barf. Different tools like `du`, `df`, `ls`, `find` etc will show varying performance too. There are even some benchmarks showing that opening a single random file can take half a second longer in a directory filled with lots of files on EXT4.

Also to add - if you're mounting a network share, there's the protocol to consider (NFS, SMB etc)

**Reply by NickM-27** [reply]
> I would like to get round to a PR to move to date-stamped folder names, but it's a fair bit of work given inconsistent approaches around the codebase plus the fact you'd have to migrate existing data for users with some careful scripting that might take 5-10 minutes to run on startup, while potentially hammering IO.

If you're going to do that I'd suggest leaving existing snapshots alone, adding a new DB field for the path to the snapshot, and then just assume when the field is null that it is in the default path.

**Reply by leccelecce** [reply]
Thanks, that makes it vastly easier. Given the number of locations the snapshot path/filename is generated I would probably pull it out into a helper method.

I can see loads of changes seem to be going into dev again, is that for a planned 0.15.1 or is it could to become 0.16?

**Reply by NickM-27** [reply]
Dev is 0.16

**Reply by leccelecce** [reply]
Ah yes, confused the open PR for 0.16 release with the merge PR closed a few days ago. I'll get round to it at some point! Great seeing all the continued fantastic improvements, incredible job :-)
